lib/spack/spack/cmd/uninstall.py

```diff
@@ -147,10 +147,10 @@ def do_uninstall(specs, force):
         try:
             # should work if package is known to spack
             packages.append(item.package)
-        except spack.repository.UnknownPackageError:
+        except spack.repository.UnknownEntityError:
             # The package.py file has gone away -- but still
             # want to uninstall.
-            spack.Package(item).do_uninstall(force=True)
+            spack.Package.uninstall_by_spec(item, force=True)
```

Comment:
@alalazo I agree -- I can try to make the interface have nice properties eventually.  There are some lifecycle issues currently that make it hard.